### ISSUE TITLE
Copied code from justshare-client -  No credits or mit license

### ISSUE THREAD

**Issue by hello898989** [subins2000/simple-peer-files#3]
Most of the code is copied/ modified from [just share-client](https://github.com/AndrewBastin/justshare-client). Even though @AndrewBastin didn't license / copyrightht it with Proper licenses with his project, it's totally unfair to take someone's code and turn it into a library and issue a non-MIT license for the same.

Legally you are allowed to do this since justshare-client is not copyrighted, but not even giving credit in the readme for the project u copied/"referred" from totally destroys the spirit of opensource.

**Comment by AndrewBastin** [comment]
Umm he did ask me if he could do it and I let him 😂

So yeah. It's cool.

P.S. The reason I didn't put up any licenses is because I don't really care about people using my code. It was just a fun experiment and that's pretty much it. If people can find use with it for actual serious efforts, then so be it ^_^

**Comment by subins2000** [comment]
> Legally you are allowed to do this since justshare-client is not copyrighte

Nope. If there is no stated LICENSE on code/github, the copyright remain solely with the author and need explicit permission for using that code. Just because there's no LICENSE mentioned doesn't mean the code in GitHub is public-domain. I did ask Andrew before making this library.

> Most of the code is copied/ modified

Let me give some context here. I was trying to find an alternative to WebTorrent in sending files in [WebDrop](https://webdrop.space). Sending files over WebTorrent is performance-heavy. IIRC, Andrew sent me his project saying that he has also made a P2P file share. So I checked that out and found the chunked file sending. I asked his permission and used his code in the initial implementation. But then, that implementation didn't work well. So I did a MASSIVE rewrite and added many more stuff. So it's incorrect to say "most of the code is copied". His project did help me to make `simple-peer-files`. What differentiates this from that mainly is that there's very little dependencies and can be used in many places. I could've erased the previous git-history after the rewrite, but I didn't. So the credit to Andew is still in the git history. Okay, I'll add it to the README too.

> issue a non-MIT license

I care about changes to this library be also FOSS so that it helps other beneficiaries of this library. Hence Mozilla Public License.
@AndrewBastin I'd recommend you and others use MPL or GPL because if the changes/derivative works are also FOSS, it'll help your project and other users too. MIT is too permissive, and sometimes some good changes will never be made into a PR, kept hidden in some commercial app and doesn't benefit the community :(
